libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
I agree - I couldn't work out exactly how it get triggers hence I gave it a generic message. But at the same time. If we do somehow reach there without triggering an exception previously  then shit would go down :/